### DISCUSSION TITLE
How to configure/constrain node version globally?

### DISCUSSION THREAD

**Discussion by tmillr** [renovatebot/renovate#19564] [Request Help]
My package is a custom js gh action (essentially a node/npm/js package in the eyes of renovate), so I think that it's probably easiest/best and most correct to have my pkg restricted to the node major that I've selected in `action.yml` (i.e. `node16`). Additionally, I'd like to ignore updates to the `engines.node` field and manage that manually instead (because, as far as I know, I have a better notion about which versions of node the gh actions runners are currently using than renovate does, and that version is not simply the latest `16.x.x`). For example, currently the node version that actually gets used when `action.yml` contains `node16` appears to be `v16.13.0`, therefore I'd like to set `package.json` -> `engines` -> `node` to `^16.13.0` and keep it there without renovate requesting me to update it to a later v16, or v18, etc. Keeping `engines.node` at `^16.13.0` (at least for now) also gives me the benefit of a central location for retrieving/determining the minimum node@16 that I need to be testing on/staying compatible with ([for example](https://github.com/tmillr/breaking-change/commit/4d90f85c85a011fab2eeda2c8269fcb16cd4bc6c)).

So my goals are essentially this:

1. I want to ignore updates to the `package.json` -> `engines` -> `node` field specifically so that I can manage that field manually instead (I'll then probably keep it at `^16.13.0` manually until node18 becomes available for gh actions and my pkg migrates to node18).
2. For anywhere else a node version might potentially appear (e.g. `package.json` -> `volta` -> `node`), I'd like it to be constrained to `^16.0.0`. So basically, totally independent from goal 1, I want to configure renovate in such a manner that it will never update a recorded node version beyond `16.x.x`.

I've attempted to accomplish this a couple different ways already (by attempting to first constrain the node version to v16 globally), but have only failed. So instead of trying further ideas/techniques and going through further trial and error (i.e. trial commits pushed directly to my master branch), I thought it might be a better idea to just ask here in hopes of getting this dealt with once and for all. Thank you!

## My Initial Attempts

https://github.com/tmillr/breaking-change/commit/6226084799d7a67065ba1129b9f1565f674bea97

> I tried this because iirc, the docs only say that `allowedVersions` is **_usually_** used as a packageRule, not that it **_must_** be (the latter of which is actually the case).

https://github.com/tmillr/breaking-change/commit/6a6a01b5f648bbcdeb2db78179936b8f2a439cd8

> I also tried this, but it appears to do nothing. The pr to upgrade to node18 is still open.

**Comment by tmillr** [comment]
bump

**Comment by viceice** [comment]
renovate won't bump ranges by default, so simply enable dashboard approval for node major updates and you should be fine.
